Adding new command `cilium ip metadata`

chowmean opened the pull request:
* Adding a new command to list down the identity metadata information for kube-apiserver policy matching. 

PR related to: https://github.com/cilium/cilium/issues/18529

```release-note
Adding cilium ip metadata command to list down the identity metadata information for kube-apisever policy matching for debugging
```

christarazi reviewed (changes requested):
A couple things

* Travis is failing with
```
docker container run --rm --workdir /src/Documentation --volume /home/travis/gopath/src/github.com/cilium/cilium/Documentation/..:/src --env READTHEDOCS_VERSION= --env SKIP_LINT= --user "2000:2000" cilium/docs-builder ./update-cmdref.sh

./check-cmdref.sh

diff --git a/Documentation/cmdref/cilium_ip.md b/Documentation/cmdref/cilium_ip.md
index 10907df68..4c94df49b 100644
--- a/Documentation/cmdref/cilium_ip.md
+++ b/Documentation/cmdref/cilium_ip.md
@@ -22,4 +22,5 @@ Manage IP addresses and associated information
 

 * [cilium](../cilium)	 - CLI
 * [cilium ip list](../cilium_ip_list)	 - List IP addresses in the userspace IPcache
+* [cilium ip metadata](../cilium_ip_metadata)	 - List ID metadata information
 
HINT: to fix this, run 'make -C Documentation update-cmdref'
```

Please run the above command and amend the commit with the changes.

* Could you update the PR description with the usage of the new command and how it looks like?

twpayne commented on cilium/cmd/ip_metadata.go:
```diff
@@ -0,0 +1,44 @@
+
+var ipMetadataCmd = &cobra.Command{
+	Use:     "metadata",
+	Aliases: []string{"md"},
```
I don't think it's necessary to add an alias for this command.

chowmean replied:
removed the alias

twpayne commented on pkg/ipcache/metadata.go:
```diff
@@ -73,6 +73,12 @@ func GetIDMetadataByIP(prefix string) labels.Labels {
 
+func GetIDMetadata() map[string]labels.Labels {
+	idMDMU.RLock()
+	defer idMDMU.RUnlock()
```
The locking here has no effect, as the function returns the map after the lock is released. For the lock to have any effect, you have to copy the map into a new map and return the copy.

chowmean replied:
@twpayne removed the locks since they have no effect. Please resolve these conversations if you think these are fine. Thanks!

christarazi replied:
@chowmean I think you still need the locks as you iterate over `identityMetadata` to create a copy of the map

Something like:

```go
func GetIDMetadata() map[string]labels.Labels {
	idMDMU.RLock()
	defer idMDMU.RUnlock()
        c := make(map[string]labels.Labels, len(identityMetadata))
        for k, v := range identityMetadata {
                 c[k] = v
        }
        return c
}
```

chowmean replied:
Added the above. Still not getting any output for ip metadata command.

chowmean committed:
Adding new command cilium ip metadata

    List state of IDMD for debugging kube-apiserver policy matching
    Documentation update for cilium ip metadata command

    Signed-off-by: Gaurav Yadav <gaurav.dev.iiitm@gmail.com>
